pkg/node/publish_unpublish.go

```diff
@@ -64,6 +94,38 @@ func (n *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 		return &csi.NodePublishVolumeResponse{}, nil
 	}
 
+	extractPodLabels := func() (map[string]string, error) {
+		volumeContext, volumeLabels := req.GetVolumeContext(), vol.ObjectMeta.GetLabels()
+		if volumeLabels == nil {
+			volumeLabels = make(map[string]string)
+		}
+
+		podName, podNs, parseErr := parseVolumeContext(volumeContext)
+		if parseErr != nil {
+			return nil, status.Errorf(codes.Internal, "failed to parse the volume context: %v", parseErr)
+		}
+
+		pod, err := utils.GetKubeClient().CoreV1().Pods(podNs).Get(ctx, podName, metav1.GetOptions{})
+		if err != nil {
+			return nil, status.Errorf(codes.NotFound, "Pod not found: %v", err)
+		}
+
+		podLabels := pod.ObjectMeta.GetLabels()
+		for k, v := range podLabels {
+			if strings.HasPrefix(k, directv1alpha1.Group+"/") {
+				volumeLabels[k] = v
+			}
+		}
+
+		return volumeLabels, nil
+	}
+
+	volLabels, vErr := extractPodLabels()
+	if vErr != nil {
```

Comment:
do not throw error if podInfoOnMount is not set. i.e. if pod name and namespace are missing